frontend/public/components/utils/breadcrumbs.ts

```diff
@@ -10,7 +10,14 @@ export const getBreadcrumbPath = (match: any, customPlural?: string) => {
     return lastNamespace === ALL_NAMESPACES_KEY
       ? `/k8s/all-namespaces/${customPlural || match.params.plural}`
       : `/k8s/ns/${match.params.ns}/${customPlural || match.params.plural}`;
+  } else if (customPlural === 'roles' || customPlural === 'rolebindings') {
```

Comment:
I'm unsure why we need this logic now. Can't this all be handled in the details page with `breadcrumbsFor`?

Reply:
No, for cluster resources (ClusterRole & ClusterRoleBinding) there is no `match.param.ns`, and without this logic would return `/k8s/cluster/...` instead of a namespaced Role/Rolebinding.

Reply:
But can't you get the namespace from `sessionStorage` in `breadscrumbFor` in those details pages?

We want to avoid special handling of specific types here.

Reply:
HI @spadgett, I kinda had that previously, but your previous [comment](https://github.com/openshift/console/pull/8010#discussion_r570251847) seemed to indicate that `sessionStorage` should be consolidated in `breadcrumbs.jsx`:

"I realize you can't use getBreadcrumbPath from breadcrumbs.ts because the path is different, but we should factor out the common logic for determining the correct namespace to avoid duplicating it [in role.jsx]. The knowledge of where it is in sessionStorage and how the breadcrumb namespace is determined should only be in one place [breadcrumbs.ts]."

Reverted so no changes to `breadcrumbs.ts` and updated `role.jsx` to:
<pre>
  const getBreadcrumbs = (modelRef, kindObj, match) => {
  const model = modelFor(modelRef);
  <b>if (!match.params.ns) {
    match.params.ns = sessionStorage.getItem(LAST_NAMESPACE_NAME_LOCAL_STORAGE_KEY);
  }</b>
  return [
    {
      name: i18next.t(model.labelPluralKey) || model.labelPlural,
      path: getBreadcrumbPath(match, model.plural),
    },
</pre>

Reply:
Code is cleaner this way :-)

Reply:
I think this is better, although we shouldn't set the `match.params.ns` value to something else.

> I kinda had that previously, but your previous comment seemed to indicate that sessionStorage should be consolidated in breadcrumbs.ts

Ideally, we do both. We keep the breadcrumb logic for role bindings in its own page, but then have a helper like `getBreadcrumbNamespace` in `breadcrumbs.ts` that keeps the session storage logic in one place there.

Reply:
> I think this is better, although we shouldn't set the `match.params.ns` value to something else.

For ClusterRole and ClusterRoleBinding there is no `match.params.ns`.

Reply:
Right, but we can't change the value of the `match` object that is passed to us from react-router. Treat it as read-only.

Reply:
Can I clone and then modify cloned match obj?